src/label_map.cpp

```diff
@@ -74,36 +74,44 @@ void LabelMap::modify_lmap(int narg, char **arg)
 
   int ntypes;
   std::vector<std::string> *labels;
+  std::unordered_map<std::string, int> *labels_map;
   const std::string tlabel(arg[0]);
   if (tlabel == "atom") {
     ntypes = natomtypes;
     labels = &typelabel;
+    labels_map = &typelabel_map;
   } else if (tlabel == "bond") {
     ntypes = nbondtypes;
     labels = &btypelabel;
+    labels_map = &btypelabel_map;
   } else if (tlabel == "angle") {
     ntypes = nangletypes;
     labels = &atypelabel;
+    labels_map = &atypelabel_map;
   } else if (tlabel == "dihedral") {
     ntypes = ndihedraltypes;
     labels = &dtypelabel;
+    labels_map = &dtypelabel_map;
   } else if (tlabel == "improper") {
     ntypes = nimpropertypes;
     labels = &itypelabel;
+    labels_map = &itypelabel_map;
   } else error->all(FLERR,"Illegal labelmap command");
 
-  int itype;
   int iarg = 1;
   while (iarg < narg) {
-    itype = utils::inumeric(FLERR,arg[iarg++],false,lmp);
+    int itype = utils::inumeric(FLERR,arg[iarg++],false,lmp);
     if (itype > ntypes)
       error->all(FLERR,"Topology type exceeds system topology type");
     std::string slabel(arg[iarg++]);
     if (isdigit(slabel[0]))
       error->all(FLERR,"Type labels cannot start with a number");
-    if (search(slabel,(*labels),ntypes) != -1)
+    if (search(slabel,(*labels_map)) != -1)
       error->all(FLERR,"Type label already exists: types labels must be unique");
-    (*labels)[itype-1] = slabel;
+    std::string &str = (*labels)[itype-1];
+    if (!str.empty()) (*labels_map).erase(str);
```

Comment:
why did you refactor the string assignment here?

Reply:
@jrgissing this is done to be compatible with vector implementation. If the `labels` at position `i` is empty means it is the first time assignment and the map also adds the same label. If the user later wants to overwrite the label e.x. with `labelmap atom i re-rename`, if I do not erase the previous label from the map, we would end up with both old and new labels in the map which is not correct.